tests/target/issue-3669/type_compressed-two.rs

```diff
@@ -0,0 +1,45 @@
+// rustfmt-version: Two
+// rustfmt-type_punctuation_density: Compressed
+
+pub fn do_something<'a, T: Trait1+Trait2+'a>(
+    &fooo: u32,
+) -> impl Foo
++Foo
```

Comment:
I think we still want to indent these `+Foo`s even when `type_punctuation_density` is set to `Compressed`.

Reply:
Indenting these `+Foo` would go against the fix for #3701: https://github.com/rust-lang/rustfmt/pull/3731/files#diff-dfd26227c4842bd3288e9b1f53f0089fR751

https://github.com/rust-lang/rustfmt/blob/ed7c0329aff1a12d340c82004b04d00645897c70/src/types.rs#L751

The `false` argument instructs `join_bounds` **not** to indent; to indent `+Foo` I would have to pass `true` instead.

BTW This non-indentation happens regardless of `type_punctuation_density`.

What should I do here ?